addons/website_sale_picking/models/sale_order.py

```diff
@@ -0,0 +1,19 @@
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _check_carrier_quotation(self, force_carrier_id=None, keep_carrier=False):
+        res = super()._check_carrier_quotation(force_carrier_id, keep_carrier)
+        if res:
+            carrier = force_carrier_id and self.env['delivery.carrier'].browse(
+                force_carrier_id) or self.carrier_id
```

Comment:
Hey, just small question, why do not you just use `self.carrier_id`, why do you browse for the `force_carrier_id` ?

Reply:
but in the parent method the selected carrier with `force_carrier_id` should already be set on the order.

Reply:
Well, after the super call, if it's not set on the order, it means it's not the applied one.